### PR TITLE
AdminLTE  IFrame

### PR DESCRIPTION
**Pull request by warquia**
| Question                | Answer
| ----------------------- | -----------------------
| Issue or Enhancement    |Enhancement
| License                 | MIT

#### What's in this PR?

**### Support the feature IFrame.** 
Question about: [https://github.com/jeroennoten/Laravel-AdminLTE/issues/917](url)

-After activating the IFrame, all Pages already assumed will be automatically loaded into a Frame.
-Compatible with pages already implemented in the traditional model without having to change any line of code

**New configuration in Config/AdminLTE:** 
    'layout_iframe' => [
        'active' => true, /*Active views in Frame*/
        'url-default' => [
            'url' => '', /*If you want to start with a page loaded Ex: Dashboard*/
            'title' => 'Dashboard', /*Title/Caption de default Page*/
        ],
        'buttons' => [
            'close'=> [
                'active' => true,
                'caption' => 'Close',
            ],
            'close-all'=> [
                'active' => true,
                'caption' => 'Close All',
            ],
            'close-all-other'=> [
                'active' => true,
                'caption' => 'Close All Other',
            ],
            'scroll-left'=> true,
            'scroll-right'=> true,
            'fullscreen'=> true,
        ],
        'captions' => [
            'no-tab' => 'No tab selected!',
            'loading' => 'Tab is loading',
        ]
    ],

#### Checklist
- [x]  I tested these changes.

### CODE REVIEW

**Comment by warquia** [comment]
Hi @Shidersz, Could you help me with this conflict on CI?

**Comment by resslinger** [comment]
@warquia in the last line of the new config are spaces. The spaces should be remove, so the ci will be run correct. :)

https://github.styleci.io/analyses/QM5wZO?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by warquia** [comment]
> @warquia in the last line of the new config are spaces. The spaces should be remove, so the ci will be run correct. :)
> 
> https://github.styleci.io/analyses/QM5wZO?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link


Thanks @resslinger

File: src/Helpers/LayoutHelper.php [hunk 1]
```diff
@@ -194,11 +197,10 @@ private static function makeSidebarClasses()
 
         // Add classes related to the "sidebar_mini" configuration.
 
-        $sidebarMiniCfg = config('adminlte.sidebar_mini', 'lg');
```
Comment (Shidersz) [inline src/Helpers/LayoutHelper.php:197]: This should not be deleted, it was part of a recent contribution... I think you need update your master branch to the latest state and then rebase your AdminLTEIFrame branch.

**Review by Shidersz — CHANGES_REQUESTED** [review]

**Comment by resslinger** [comment]
Is the menu prepared for the iframe, so that links will be open in the iframe? Or changes are needed? @warquia @Shidersz

**Comment by warquia** [comment]
> Is the menu prepared for the iframe, so that links will be open in the iframe? Or changes are needed? @warquia @Shidersz

Good night @resslinger, no changes will be necessary, when activated all menu links will be rederized within the IFrame if no refactoring by users,

**Comment by warquia** [comment]
> @resslinger Não tenho certeza, não tenho nenhuma experiência com o recurso IFrame do AdminLTE. Precisamos fazer uma análise / teste em profundidade antes da fusão. Precisamos ter certeza de que isso não entre em conflito com outras coisas, porque custa muito tempo para colocar o pacote em um ponto estável com os componentes e AdminLTE v3.1.

**Comment by warquia** [comment]
> > @resslinger Não tenho certeza, não tenho nenhuma experiência com o recurso IFrame do AdminLTE. Precisamos fazer uma análise / teste em profundidade antes da fusão. Precisamos ter certeza de que isso não entre em conflito com outras coisas, porque custa muito tempo para colocar o pacote em um ponto estável com os componentes e AdminLTE v3.1.

I agree with you @Shidersz .
Tomorrow morning I will update the sources

**Comment by Shidersz** [comment]
Thanks @warquia . I'm currently out of free time, but I will try to review this new feature as soon as I can. By the time, try to use [git rebase](https://git-scm.com/docs/git-rebase) to update this branch to the last commit of the master. You may need to configure this project as a remote of your local git repo, so you can pull changes made on the `master` of this repo and sync your master branch. Read [here](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork) for details...

**Review by resslinger — APPROVED** [review]

**Comment by Shidersz** [comment]
Why this was merged? It was not tested or reviewed and there is no documentation on the wiki. I'm sure there are a lot of fix and improvements to do on the configuration format and the layout... Personally, I will have waited some time...

**Comment by resslinger** [comment]
I need this in a other project, so I work on it. It is not published as a release, so we can improve and fix that.

**Comment by Shidersz** [comment]
Ok, what is the main benefit of using the IFrame plugin? I'm still not sure...

**Comment by resslinger** [comment]
You can test it on the demo site: https://adminlte.io/themes/v3/iframe.html

With the iframe feature, all links will be open on the same site. So you can work with more open sites in one window. :)

**Comment by Shidersz** [comment]
Ok, I believe, for the amount of configuration options available, it should be moved to a different section other than layout.

**Comment by resslinger** [comment]
I have reverted the merge to master branch and created the branch "add-feature-iframe". In this branch, we can optimize and change the iframe integration.

**Comment by Shidersz** [comment]
@resslinger Thanks you, I will give this a try soon...

**Pull request merged**